extensions/azurecore/src/azureResource/providers/databaseServer/databaseServerService.ts

```diff
@@ -14,20 +17,46 @@ export interface DbServerGraphData extends GraphData {
 	};
 }
 
-export const serversQuery = `where type == "${azureResource.AzureResourceType.sqlServer}"`;
+export interface DbSynapseGraphData extends GraphData {
+	properties: {
+		connectivityEndpoints: { sql: string };
+		managedResourceGroupName: string;
+		sqlAdministratorLogin: string;
+	};
+}
 
-export class AzureResourceDatabaseServerService extends ResourceServiceBase<DbServerGraphData, azureResource.AzureResourceDatabaseServer> {
+export class AzureResourceDatabaseServerService implements IAzureResourceService<azureResource.AzureResourceDatabaseServer> {
 
 	protected get query(): string {
-		return serversQuery;
+		return sqlServersQuery;
 	}
 
-	protected convertResource(resource: DbServerGraphData): azureResource.AzureResourceDatabaseServer {
+	public async getResources(subscriptions: azureResource.AzureResourceSubscription[], credential: ServiceClientCredentials, account: AzureAccount): Promise<azureResource.AzureResourceDatabaseServer[]> {
+		const convertedResources: azureResource.AzureResourceDatabaseServer[] = [];
+		const resourceClient = new ResourceGraphClient(credential, { baseUri: account.properties.providerSettings.settings.armResource.endpoint });
+		let graphResources: (DbSynapseGraphData | DbServerGraphData)[] = await queryGraphResources<DbServerGraphData>(resourceClient, subscriptions, this.query);
```

Comment:
Type should just be `DbServerGraphData[]` - and rename this var to `serverGraphResources`